### PR TITLE
docs: Update shared service annotation docs

### LINKED ISSUE #19273: docs: Update global service user docs
As followed up action of https://github.com/cilium/cilium/pull/19042, related user docs must be updated https://docs.cilium.io/en/latest/gettingstarted/clustermesh/services/#load-balancing-with-global-services

To avoid any confusion, we can expand example with actual output when sending requests from each cluster.

### PR DESCRIPTION
**Pull request by sayboras**
This commit is to update related docs for io.cilium/shared-service
annotation, along with steps and expected behavior.

Fixes #19273

Signed-off-by: Tam Mach <tam.mach@isovalent.com>

### CODE REVIEW

File: Documentation/gettingstarted/clustermesh/services.rst [hunk 1]
```diff
@@ -24,14 +24,12 @@ Cilium will automatically perform load-balancing to pods in both clusters.
 .. literalinclude:: ../../../examples/kubernetes/clustermesh/global-service-example/rebel-base-global-shared.yaml
   :language: YAML
 
-Load-balancing Only to a Remote Cluster
-#######################################
+Disabling Global Service sharing
```
Comment (qmonnet) [inline Documentation/gettingstarted/clustermesh/services.rst]:
```suggestion
Disabling Global Service Sharing
```

**Review by qmonnet — APPROVED** [review]
Thanks for following up!

**Comment by pchaigno** [comment]
Don't we need a review from @cilium/sig-clustermesh here?

**Comment by sayboras** [comment]
> Don't we need a review from @cilium/sig-clustermesh here?

From the previous, I assume that clustermesh team is aware of changes and agreed with the approach. But yup, no harm to have another look at docs. Honestly, writing docs is much harder than code at least to me, so much appreciate all the help.

**Review by raymonddejong — APPROVED** [review]
Good addition to explain the shared-service=false annotation.

**Review by joestringer — APPROVED** [review]

**Pull request merged**